From ae2e0c9708aeb4a8fb301a1a4016cec7b05fc142 Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Tue, 2 Jan 2018 06:55:41 +0200
Subject: [PATCH] lib/Module.mk: don't install dynamic library when disabled

Do not attempt to install the dynamic library when build of that library
is disabled. Fixes the following installation error:

install -m 755 lib/libi2c.so.0.1.0 .../target/usr/lib
install: cannot stat 'lib/libi2c.so.0.1.0': No such file or directory
lib/Module.mk:90: recipe for target 'install-lib' failed

Cc: Angelo Compagnucci <angelo@amarulasolutions.com>
Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---

Patch status: posted upstream
https://marc.info/?l=linux-i2c&m=151486925730634&w=2

 lib/Module.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/Module.mk b/lib/Module.mk
index 2ebc70d76ed9..c492961e0a4a 100644
--- a/lib/Module.mk
+++ b/lib/Module.mk
@@ -88,9 +88,11 @@ clean-lib:
 
 install-lib: $(addprefix $(LIB_DIR)/,$(LIB_TARGETS))
 	$(INSTALL_DIR) $(DESTDIR)$(libdir)
+ifeq ($(BUILD_DYNAMIC_LIB),1)
 	$(INSTALL_PROGRAM) $(LIB_DIR)/$(LIB_SHLIBNAME) $(DESTDIR)$(libdir)
 	$(LN) $(LIB_SHLIBNAME) $(DESTDIR)$(libdir)/$(LIB_SHSONAME)
 	$(LN) $(LIB_SHSONAME) $(DESTDIR)$(libdir)/$(LIB_SHBASENAME)
+endif
 ifeq ($(BUILD_STATIC_LIB),1)
 	$(INSTALL_DATA) $(LIB_DIR)/$(LIB_STLIBNAME) $(DESTDIR)$(libdir)
 endif
-- 
2.15.1

